### ISSUE TITLE
GeoPackage Zip File Reader Error: Attempt to read past end of input

### ISSUE THREAD

**Issue by wipfli** [onthegomap/planetiler#556]
I would like to use a zip file containing many GeoPackage files.

The profiles looks like this:

```java
public class SwissMap implements Profile {

  @Override
  public void processFeature(SourceFeature sourceFeature, FeatureCollector features) {

    if ("Snow-4".equals(sourceFeature.getSource())) {
      features.polygon("landcover")
        .setMinZoom(1)
        .setMaxZoom(1)
        .setAttr("kind", "Snow")
        .setPixelTolerance(0.2);
    }
  }

  static void run(Arguments args) throws Exception {
    String area = args.getString("area", "geofabrik area to download", "monaco");
    Planetiler.create(args)
      .setProfile(new SwissMap())
      .addGeoPackageSource("EPSG:4326", "Snow-4", Path.of("data", "sources", "Snow-resolution-4.zip"), "")
      .run();
  }
}

```

And here is the `Snow-resolution-4.zip` file: 
[Snow-resolution-4.zip](https://github.com/onthegomap/planetiler/files/11287641/Snow-resolution-4.zip)

When running this profile, I get the following error:

```
0:00:03 INF [Snow-4] - 
0:00:03 INF [Snow-4] - Starting...
Worker Snow-4_read-1 died
org.locationtech.jts.io.ParseException: Attempt to read past end of input
        at org.locationtech.jts.io.ByteOrderDataInStream.read(ByteOrderDataInStream.java:142)
        at org.locationtech.jts.io.ByteOrderDataInStream.readByte(ByteOrderDataInStream.java:93)
        at org.geotools.geometry.jts.WKBReader.readGeometry(WKBReader.java:176)
        at org.geotools.geometry.jts.WKBReader.read(WKBReader.java:170)
        at org.geotools.geometry.jts.WKBReader.read(WKBReader.java:154)
        at com.onthegomap.planetiler.reader.GeoPackageReader.readFeatures(GeoPackageReader.java:143)
        at com.onthegomap.planetiler.reader.SourceFeatureProcessor.lambda$readPaths$3(SourceFeatureProcessor.java:174)
        at com.onthegomap.planetiler.worker.WorkerPipeline$Builder.lambda$addWorker$0(WorkerPipeline.java:252)
        at com.onthegomap.planetiler.worker.Worker.lambda$new$0(Worker.java:41)
        at com.onthegomap.planetiler.worker.Worker.lambda$new$1(Worker.java:68)
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Exception in thread "main" com.onthegomap.planetiler.util.Exceptions$FatalPlanetilerException: java.util.concurrent.ExecutionException: com.onthegomap.planetiler.util.Exceptions$FatalPlanetilerException: org.locationtech.jts.io.ParseException: Attempt to read past end of input
        at com.onthegomap.planetiler.util.Exceptions.throwFatalException(Exceptions.java:29)
        at com.onthegomap.planetiler.stats.ProgressLoggers.await(ProgressLoggers.java:367)
        at com.onthegomap.planetiler.stats.ProgressLoggers.awaitAndLog(ProgressLoggers.java:352)
        at com.onthegomap.planetiler.worker.WorkerPipeline.awaitAndLog(WorkerPipeline.java:62)
        at com.onthegomap.planetiler.reader.SourceFeatureProcessor.processFiles(SourceFeatureProcessor.java:130)
        at com.onthegomap.planetiler.reader.SourceFeatureProcessor.processFiles(SourceFeatureProcessor.java:65)
        at com.onthegomap.planetiler.reader.GeoPackageReader.process(GeoPackageReader.java:102)
        at com.onthegomap.planetiler.Planetiler.lambda$addGeoPackageSource$4(Planetiler.java:387)
        at com.onthegomap.planetiler.Planetiler.lambda$ifSourceUsed$9(Planetiler.java:841)
        at com.onthegomap.planetiler.Planetiler.run(Planetiler.java:720)
        at com.onthegomap.planetiler.examples.SwissMap.run(SwissMap.java:189)
        at com.onthegomap.planetiler.examples.SwissMap.main(SwissMap.java:173)
Caused by: java.util.concurrent.ExecutionException: com.onthegomap.planetiler.util.Exceptions$FatalPlanetilerException: org.locationtech.jts.io.ParseException: Attempt to read past end of input
        at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
        at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
        at com.onthegomap.planetiler.stats.ProgressLoggers.await(ProgressLoggers.java:361)
        ... 10 more
Caused by: com.onthegomap.planetiler.util.Exceptions$FatalPlanetilerException: org.locationtech.jts.io.ParseException: Attempt to read past end of input
        at com.onthegomap.planetiler.util.Exceptions.throwFatalException(Exceptions.java:29)
        at com.onthegomap.planetiler.worker.Worker.lambda$new$1(Worker.java:76)
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.locationtech.jts.io.ParseException: Attempt to read past end of input
        at org.locationtech.jts.io.ByteOrderDataInStream.read(ByteOrderDataInStream.java:142)
        at org.locationtech.jts.io.ByteOrderDataInStream.readByte(ByteOrderDataInStream.java:93)
        at org.geotools.geometry.jts.WKBReader.readGeometry(WKBReader.java:176)
        at org.geotools.geometry.jts.WKBReader.read(WKBReader.java:170)
        at org.geotools.geometry.jts.WKBReader.read(WKBReader.java:154)
        at com.onthegomap.planetiler.reader.GeoPackageReader.readFeatures(GeoPackageReader.java:143)
        at com.onthegomap.planetiler.reader.SourceFeatureProcessor.lambda$readPaths$3(SourceFeatureProcessor.java:174)
        at com.onthegomap.planetiler.worker.WorkerPipeline$Builder.lambda$addWorker$0(WorkerPipeline.java:252)
        at com.onthegomap.planetiler.worker.Worker.lambda$new$0(Worker.java:41)
        at com.onthegomap.planetiler.worker.Worker.lambda$new$1(Worker.java:68)
```

Is there anything obvious which I am doing wrong? Like for example, can a GeoPackage zip file contain more than one file?

Thanks for any help with this!

**Comment by wipfli** [comment]
If I delete all but one .gpkg files from the zip archive, it works.

With the following zip file I don't get an error: 
[Snow-resolution-4.zip](https://github.com/onthegomap/planetiler/files/11288008/Snow-resolution-4.zip)

**Comment by wipfli** [comment]
With two non-empty gpkg files it works. The following zip archive is fine: 
[Snow-resolution-4.zip](https://github.com/onthegomap/planetiler/files/11288023/Snow-resolution-4.zip)

**Comment by wipfli** [comment]
OK looks like indeed empty geopackage files in a zip archive make planetiler crash.

This file produces the error and it has a single empty gpkg inside: 
[Snow-resolution-4.zip](https://github.com/onthegomap/planetiler/files/11288050/Snow-resolution-4.zip)

**Comment by wipfli** [comment]
I can work around this by filtering my data for empty files. But should we add a check or something in the reader and just ignore empty files?

**Comment by phanecak-maptiler** [comment]
I agree that maybe this could be handled better but still: bad input -> bad output. And fast noisy crash is usually quite good: you know you have bad input right away, no need to troubleshoot more or less obvious errors in a result later on.

Side note: As a coincidence, I got "_Attempt to read past end of input_" just yesterday. File in question was NOT empty, as it was working fine in QGIS. Since that particular file was not my primary concern, I simply used some other file, hence I did not pursue further the root cause of that exception.
